src/hooks/useFetch.ts

```diff
@@ -0,0 +1,8 @@
+import { useVersion } from "hooks";
+
+// useFetch wraps the native Deno fetch api and inserts a User-Agent header
+export default function useFetch(input: string | URL | Request, init?: RequestInit | undefined): Promise<Response> {
+    const requestInit = init ?? {} as RequestInit
+    requestInit.headers = { ...requestInit.headers, "User-Agent": `tea-cli/${useVersion()}` }
```

Comment:
Can we go with `tea/cli/version` or is that violating expectations for parsers of the user-agent?

Maybe `tea (cli)` is more typical.

Either way I'm not fond of the dash here.

Reply:
I think `tea(cli)/version` is probably fine if a bit atypical.  

You can see a bunch of examples here:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent

Reply:
yeah you're right. Parenthesis are used for platform info not variants.

yeah, back to `tea-cli/version` I agree with your original assessment lol

Reply:
Or perhaps `tea.cli/version` which we've used elsewhere sometimes.

Reply:
tea.cli it is